libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -643,3 +648,36 @@ private val fakeDirectionsRoute: DirectionsRoute by lazy {
         .legs(fakeLegs)
         .build()
 }
+
+private fun DataRef.asReader(): Reader =
+    InputStreamReader(ByteBufferBackedInputStream(buffer))
+
+private class ByteBufferBackedInputStream(
```

Comment:
I don't test it directly, but we have tests that create routes through the data ref. Do you see any scenarios rather than serialise a few routes using this input stream to add?

Reply:
However we don't check all the property of that routes. Other tests won't fail if one of the property won't be serialised. I will add a test that compares all the properties

Reply:
Added `navigation routes serialised from string, data_ref, and model are equals` test

Reply:
We can also make this class non-private and test it directly with corner cases (empty buffer, we have to read multiple times, the last read is partial, etc.).

Reply:
Not a blocker, approving.